tree/dataframe/src/RDFUtils.cxx

```diff
@@ -48,22 +48,18 @@ namespace RDF {
 /// exception is thrown.
 /// References and pointers are not supported since those cannot be stored in
 /// columns.
+/// Note that this function will take a lock and may be a potential source of
+/// contention in multithreaded execution.
 const std::type_info &TypeName2TypeID(const std::string &name)
 {
-   if (auto c = TClass::GetClass(name.c_str())) {
-      if (!c->GetTypeInfo()) {
-         std::string msg("Cannot extract type_info of type ");
-         msg += name;
-         msg += ".";
-         throw std::runtime_error(msg);
-      }
-      return *c->GetTypeInfo();
-   } else if (name == "char" || name == "Char_t")
+   if (name == "float" || name == "Float_t")
```

Comment:
It is not clear if using something like:
```
static std::unordered_map<std::string, std::type_info&> gName2TypeInfo = { { "int", typeid(int) } , { "Int_t_", typeid(int>}, etc ... };
auto res = gName2TypInfo.find(name);
if (res != gName2TypInfo.end())
   return res.second;
if (auto c = TClass::GetClass(name.c_str())) {
   ....
```
will be more efficient or not.

Reply:
Well, even then, it's only faster by a small margin apparently https://quick-bench.com/q/oLQdf2fZfS9WJjXr0h-udcV7XAw

Reply:
Here are updated quickbench runs with an extra sanity check (`DoNotOptimize` called on the returned string from the benchmark functions).

* using only `typeid(bool)` as input (that's the last in the chain of if-else) https://quick-bench.com/q/IysbUjZI95QrvuUoZEG8VxioeoY
* using randomly generated typeids to homogenize access to map values and if-else branches https://quick-bench.com/q/w0Qx0Ml_ZdYHGmGC6A0U1k-OVrA

Reply:
I don't know whether it matters or not but we have both a function that does `{ typeid -> string }` and `{ string -> typeid }`.  If I read correctly we have benchmarks only for the former and not for the later.

Reply:
Sure, the benchmark should be equivalent, I would turn the map key<->value for that case. I can make a quick bench and copy here

Reply:
Updated benchmarks

* https://quick-bench.com/q/q3kA2rw6kz0Lf-VCErEOk7IbbS8 : 4-way comparison of the `TypeName2TypeID` function, implemented with if-else list or with map lookup, using just `typeid(bool)` as input or generating the type_info randomly
* https://quick-bench.com/q/c1rnNvJHQgXtoZNLGwrZMFuSU-o : 4-way comparison of the `TypeID2TypeName` function, implemented with if-else list or with map lookup, using just `typeid(bool)` as input or generating the type_info randomly

In all cases, it seems that the implementation with map is at least as fast, sometimes can be faster.

Reply:
As a side note it is impressive how well the branch prediction is working (how much faster `WithIfElseBool` is compared to `WithIfElseRndm`)